content/general/concepts/abstract-factory/abstract-factory.md

```diff
@@ -0,0 +1,396 @@
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+Next is the Silver-level implementation of the `CurrentAccount`:
+
+```java
+public class SilverPersonal extends CurrentAccount {
+  // The Silver Personal has a mid max overdraft limit
+  public static final int MAX_OVERDRAFT = 1200;
+
+  public SilverPersonal(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    super(accountHolder, initBalance, agreedOverdraftLimit);
+  }
+
+  // Concrete overridden method
+  @Override
+  public void increaseOverdraft(int requestedLimit) {
+    if (requestedLimit < MAX_OVERDRAFT) {
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+Here is the Bronze-level implementation:
+
+```java
+public class BronzePersonal extends CurrentAccount {
+  // The Bronze Personal has a low max overdraft limit
+  public static final int MAX_OVERDRAFT = 500;
+
+  public BronzePersonal(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    super(accountHolder, initBalance, agreedOverdraftLimit);
+  }
+
+  // Concrete overridden method
+  @Override
+  public void increaseOverdraft(int requestedLimit) {
+    if (requestedLimit < MAX_OVERDRAFT) {
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+And finally, the Credit Builder:
+
+```java
+public class CreditBuilder extends CurrentAccount {
+  // The Credit Builder does not allow for an overdraft
+  public static final int MAX_OVERDRAFT = 0;
+
+  public CreditBuilder(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    super(accountHolder, initBalance, agreedOverdraftLimit);
+  }
+
+  // Concrete overridden method
+  @Override
+  public void increaseOverdraft(int requestedLimit) {
+    if (requestedLimit < MAX_OVERDRAFT) {
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+In each concrete current account, a `static` variable (`MAX_OVERDRAFT`) has been provided and constructed with the correct value. This is then used in the `.increaseOverdraft()` method to ensure this limit is not exceeded.
+
+Below provides an abstract model `SavingAccount` class for its concrete sub-classes to be based on:
+
+```java
+public abstract class SavingAccount {
+  private final String accountUID = UUID.randomUUID().toString();
+  private final Customer accountHolder;
+  private final LocalDate dateOpened;
+  private double balance;
+
+  public SavingAccount(Customer accountHolder, LocalDate dateOpened, double initBalance) {
+    this.accountHolder = accountHolder;
+    this.dateOpened = dateOpened;
+    this.balance = initBalance;
+  }
+
+  public void addFunds(final int amount) {
+    balance += amount;
+  }
+
+  public void withdrawFunds(final int amount) {
+    balance -= amount;
+  }
+
+  // Enforces concrete classes to override this method
+  public abstract void addInterest();
+
+  // Getters
+}
+```
+
+**Note:** Getters have been excluded from the above code snippet.
```

Comment:
This can probably just be incorporated into the code since notes are meant for extra information a reader should take important notes of.